A72-open-telemetry-tracing.md

```diff
@@ -0,0 +1,428 @@
+
+GlobalInterceptors.setInterceptors(
+    Arrays.asList(otModule.getClientTracingInterceptor()),
+    Arrays.asList(otModule.getServerTracerFactory()));
+
+```
+
+#### C++
+In C++, it will be a method that mirrors OpenCensus API.
+
+TODO: update C++ API.
+
+```C++
+// Enable OpenTelemetry based tracing. Similar to
+// RegisterOpenCensusPlugin(). TracerProvider is configured via sdk separately.
+void RegisterOpenTelemetryTracingPlugin();
+
+```
+
+#### Go
+In Go, the OpenTelemetry stream tracers and interceptors will be provided for users to install.
+
+TODO: add Go API.
+
+### Migrate to OpenTelemetry: cross-process networking concerns
+When clients first introduce gRPC OpenTelemetry, for the time window when the 
+gRPC client and server have mixed plugins of OpenTelemetry and OpenCensus, 
+with `grpcTraceBinPropagator` users can do migration easily with the compatibility
+guaranteed. It is encouraged to use `grpcTraceBinPropagator` that propagates
+`grpc-trace-bin` header for migration because of the following advantages:
+* Simplified migration path, no migration phase deployments.
+* Binary header is more efficient.
+
+A binary formatter implementation for OpenTelemetry is needed in each language, 
+which can be similar to the OpenCensus implementation. Go already has community 
+support for that.
+
+After migration period, users have the flexibility to switch to other propagators.
+OpenTelemetry and its extension packages support multiple text map propagators, 
+e.g. W3C trace context or b3. The gRPC OpenTelemetry API allows specifying 
+multiple propagators: either public standard ones or custom propagators that 
+implement the OpenTelemetry propagators API interface. The API composites the 
+propagators and gRPC puts all the propagator data into the wire through metadata. 
+This allows users to easily migrate a group of applications with an old propagator to 
+a new propagator. An example migration path can be:
+1. Configure server to accept both old and new propagators.
+2. Configure the client with the desired new propagators and to drop the old propagator.
+3. Make the server only accept the new propagators and complete the migration.
+
+### Migrate to OpenTelemetry: in binary
+The OpenCensus [shim](https://github.com/open-telemetry/opentelemetry-java/tree/main/opencensus-shim)
+(currently available in Java, Go, Python) allows binaries that have a mix of 
+OpenTelemetry and OpenCensus dependencies to export trace spans from both frameworks,
+and keep the correct parent-child relationship. This is the recommended approach to migrate
+to OpenTelemetry in one binary gradually. Note that the in-binary migration and 
+cross-cutting concerns migration can be done in parallel.
+
+The shim packages that bridge two libraries works as follows, considering the
+following migration scenarios example: 
+
+`|-- Application - Configured OpenCensus ------------------------------- |`\
```

Comment:
Put each of these in a single code block. I think that'll make it draw nicer.

\`\`\`
Bunch
of
text
\`\`\`